### PR TITLE
feat/tokens: Update and use token checks.

### PR DESCRIPTION
**Pull request by joshuef**
Remove old app perm checks.
Update error messages for access denied
Add no token tests

### CODE REVIEW

File: src/client_handler.rs [hunk 1]
```diff
@@ -447,7 +447,9 @@ impl ClientHandler {
     /// Verifies token validity for a given PublicId
     fn is_valid_token_for_app(&self, client_id: &ClientPublicId, token: &AuthToken) -> bool {
         let public_id = &PublicId::Client(client_id.clone());
-        token.is_valid_for_public_key(&public_id.public_key()).is_ok()
+        token
+            .is_valid_for_public_key(&public_id.public_key())
+            .expect("Error validating token")
```
Comment (Yoga07) [inline src/client_handler.rs]: We should not unwrap/expect in production code. We should handle this error by returning `false` or by some other way if it is an `Err`.

File: tests/common/mod.rs [hunk 2]
```diff
@@ -177,14 +178,26 @@ impl Environment {
...
+        let mut app = TestApp::new_disconnected(&mut self.rng, owner_full_id, perms);
         self.establish_connection(&mut app);
         app
     }
 
-    pub fn new_disconnected_app(&mut self, owner: ClientPublicId) -> TestApp {
-        TestApp::new_disconnected(&mut self.rng, owner)
+    pub fn new_disconnected_app(
+        &mut self,
+        owner: &FullId,
```
Comment (Yoga07) [inline tests/common/mod.rs:193]: Why did we change the `owner` arg `FullId`? Just curious.

**Review by Yoga07 — CHANGES_REQUESTED** [review]
Just one minor change needed. Otherwise all good! :+1:

File: tests/common/mod.rs (hunk 2 above)
Comment (joshuef) [reply tests/common/mod.rs:193]: Ah, because we need the sign capability for signing the token when we instantiate the app  (at `new_disconnected`)

**Review by Yoga07 — APPROVED** [review]
Nice one! :+1:

**Pull request merged**